systemvm/debian/opt/cloud/bin/cs/CsDhcp.py

```diff
@@ -58,17 +58,11 @@ def process(self):
         if self.conf.commit():
             restart_dnsmasq = True
 
-        if self.cloud.commit():
-            restart_dnsmasq = True
-
-        if self.dhcp_leases.commit():
-            restart_dnsmasq = True
-
+        self.dhcp_hosts.commit()
+        self.dhcp_leases.commit()
         self.dhcp_opts.commit()
 
-        if restart_dnsmasq:
-            self.delete_leases()
-
+        self.delete_leases()
         self.write_hosts()
```

Comment:
@weizhouapache can you explain what this line does? Isn't deleting leases going to cause any regression?

Reply:
@rhtyd 
the method delete_leases was mainly introduced by #3351, which is used to remove entries from dnsmasq.leases file for VMs which have  been removed (no impact on all existing vms).

in my opinion, it will not cause any regression. the only issue I see is that execution time will be increased few milliseconds if stop/start a vm(currently delete_leases is not triggered).
 https://github.com/apache/cloudstack/blob/737f50d133c918f1f61580c1c23ebf37b12a3799/systemvm/debian/opt/cloud/bin/cs/CsDhcp.py#L134-L154

Reply:
@rhtyd anyway, I added a new commit to address your comment.

leases will be deleted only when one of config files is changed.

Reply:
Great thanks.